swatch-contracts/src/main/java/com/redhat/swatch/contract/model/MeasurementMetricIdTransformer.java

```diff
@@ -78,4 +84,21 @@ public void translateContractMetricIdsToSubscriptionMetricIds(SubscriptionEntity
       throw new ContractsException(ErrorCode.UNHANDLED_EXCEPTION, e.getMessage());
     }
   }
+
+  private void checkForUnsupportedMetrics(List<Metric> metrics, SubscriptionEntity subscription) {
+    if (metrics.size() != subscription.getSubscriptionMeasurements().size()
```

Comment:
should we compare the metrics by values? This condition can still pass if the metric values are different and the size of the list is same.

Reply:
Yes, we should as the metrics in the method is a set, so all i'm doing is check if the metricId of the measurement is the right type, so it wouldn't remove the measurement as long as it has the correct metric assigned to the product.

Reply:
As we spoke last Friday this has to be fixed. Since if we have subscription with Sockets and Cores then Contract with Sockets and Instance-hours then it won't remove the metric id that are not matching.